base/array.jl

```diff
@@ -813,6 +813,9 @@ function getindex(A::Array, I::UnitRange{Int})
     end
     return X
 end
+
+getindex(a::Array, r::AbstractUnitRange{Bool}) = getindex(a, to_index(r))
```

Comment:
Perhaps you could add a comment here to clarify that this carries out Logical Indexing

Reply:
Added that.